ci/unit_tests/functions_deploy/main_test.py

```diff
@@ -0,0 +1,198 @@
+            functionRespJson = functionResp.json()
+            assert pythonVersion == functionRespJson['majorVersion']
+
+
+    def test_functionsInZip(self):
+        """Tests if functions_deploy can handle function in zip file."""
+        # prepare zip file
+        dirForZip = os.path.join(self.dataBasePath, "outputs", "pythonZip")
+        BaseTestCaseCapture.createFolder(dirForZip)
+
+        with zipfile.ZipFile(os.path.join(dirForZip, 'testFunc.zip'), 'w') as functionsZip:
+            for fileToZip in os.listdir(os.path.join(self.dataBasePath, 'zip_functions')):
+                functionsZip.write(os.path.join(self.dataBasePath, 'zip_functions', fileToZip), fileToZip)
+
+        #upload zip file
+        params = ['--cloudfunctions_username', os.environ['CLOUD_FUNCTIONS_USERNAME'],
+                  '--cloudfunctions_password', os.environ['CLOUD_FUNCTIONS_PASSWORD'],
+                  '--cloudfunctions_package', self.package, '--cloudfunctions_namespace', self.urlNamespace,
+                  '--cloudfunctions_url', self.cloudFunctionsUrl, '--common_functions', [dirForZip]]
+
+        self.t_noException([params])
+        self.packageCreated = True
+
+        # call function and check if sub-function from non-main file was called
+        functionCallUrl = self.cloudFunctionsUrl + '/' + self.urlNamespace + '/actions/' + self.package + \
+            '/testFunc?blocking=true&result=true'
+
+        functionResp = requests.post(functionCallUrl, auth=(os.environ['CLOUD_FUNCTIONS_USERNAME'],
+                                                            os.environ['CLOUD_FUNCTIONS_PASSWORD']),
+                                     headers={'Content-Type': 'application/json',
+                                              'accept': 'application/json'},
+                                     data="{}")
+
+        assert functionResp.status_code == 200
+        functionRespJson = functionResp.json()
+        assert "String from helper function" == functionRespJson['test']
+
+
+    def test_badArgs(self):
+        """Tests some basic common problems with args."""
+        self.t_unrecognizedArgs([['--nonExistentArg', 'randomNonPositionalArg']])
+        self.t_exitCode(1, [[]])
+
+        completeArgsList = ['--cloudfunctions_username', os.environ['CLOUD_FUNCTIONS_USERNAME'],
+                            '--cloudfunctions_password', os.environ['CLOUD_FUNCTIONS_PASSWORD'],
+                            '--cloudfunctions_package', self.package,
+                            '--cloudfunctions_namespace', self.urlNamespace,
+                            '--cloudfunctions_url', self.cloudFunctionsUrl,
+                            '--common_functions', self.dataBasePath]
+
+        for argIndex in range(len(completeArgsList)):
+            if not completeArgsList[argIndex].startswith('--'):
+                continue
+            paramName = completeArgsList[argIndex][2:]
+
+            missingArgsList = []
```

Comment:
From my understanding this more like `providedArgsList` and `paramName` is the `missingParam`

Reply:
changed